tests/test_functional_archive_logs.py

```diff
@@ -0,0 +1,200 @@
+"""
+    test_functional_archive_logs.py
+    Test archiving schedule_log entries into schedule_log_historical
+"""
+
+
+from unittest import result
+import pytest
+import os
+import datetime
+import psycopg2
+
+from cicada.commands import archive_schedule_log
+
+
+@pytest.fixture(scope="session", autouse=True)
+def get_env_vars():
+    """get_env_vars"""
+
+    pytest.cicada_home = os.environ.get("CICADA_HOME")
+
+    pytest.db_host = os.environ.get("DB_POSTGRES_HOST")
+    pytest.db_port = os.environ.get("DB_POSTGRES_PORT")
+    pytest.db_user = os.environ.get("DB_POSTGRES_USER")
+    pytest.db_pass = os.environ.get("DB_POSTGRES_PASS")
+
+    pytest.db_test = f"pytest_{datetime.datetime.now().strftime('%Y%m%d_%H%M%S_%f')}"
+
+
+@pytest.fixture()
+def db_setup(get_env_vars):
+    """db_setup"""
+
+    # Create the test_db
+    pg_conn = psycopg2.connect(
+        host=pytest.db_host,
+        port=pytest.db_port,
+        user=pytest.db_user,
+        password=pytest.db_pass,
+        database="postgres",
+    )
+    pg_conn.autocommit = True
+    pg_cur = pg_conn.cursor()
+    pg_cur.execute(f"CREATE DATABASE {pytest.db_test}")
+
+    # Create test_db structure
+    test_conn = psycopg2.connect(
+        host=pytest.db_host,
+        port=pytest.db_port,
+        user=pytest.db_user,
+        password=pytest.db_pass,
+        database=pytest.db_test,
+    )
+    test_conn.autocommit = True
+    test_cur = test_conn.cursor()
+    test_cur.execute(
+        open(f"{pytest.cicada_home}/setup/schema.sql", "r", encoding="utf-8").read()
+    )
+    test_cur.close()
+    test_conn.close()
+
+
+def query_test_db(query):
+    """Run and SQL query in a postgres database"""
+    rows = []
+    with psycopg2.connect(
+        host=pytest.db_host,
+        port=pytest.db_port,
+        user=pytest.db_user,
+        password=pytest.db_pass,
+        database=pytest.db_test,
+    ) as conn:
+        conn.set_session(readonly=False, autocommit=True)
+        with conn.cursor() as cur:
+            cur.execute(query)
+            if cur.rowcount > 0 and cur.description:
+                rows = cur.fetchall()
+    return rows
+
+
+def test_test_db_setup(db_setup):
+    """test_test_db_setup"""
+    query_result = query_test_db("SELECT 1")
+
+    assert query_result == [(1,)]
+
+
+def test_create_dummy_servers():
```

Comment:
Why is this a test? this is more of a setup step that prepare data fixtures for main relevant test `test_archive_schedules`